ethers-providers/src/transports/common.rs

```diff
@@ -103,6 +103,16 @@ pub enum Authorization {
     Bearer(String),
 }
 
+impl Authorization {
+    pub fn basic(username: impl ToString, password: impl ToString) -> Self {
```

Comment:
Use `impl Into<String>` because calling `String::to_strig` effectively clones, whereas `into` does not
```suggestion
    pub fn basic(username: impl Into<String>, password: impl Into<String>) -> Self {
```